src/autosuggest/__tests__/virtual-scroll.test.tsx

```diff
@@ -0,0 +1,66 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { render } from '@testing-library/react';
+import Autosuggest, { AutosuggestProps } from '../../../lib/components/autosuggest';
+import createWrapper from '../../../lib/components/test-utils/dom';
+
+const defaultProps: AutosuggestProps = {
+  options: [
+    { value: '1', label: 'One' },
+    { value: '2', lang: 'Two' },
+    { value: '3', lang: 'Three' },
+  ],
+  virtualScroll: true,
+  enteredTextLabel: () => 'Use value',
+  value: '',
+  onChange: () => {},
+};
+
+function renderWithWrapper(ui: React.ReactElement) {
+  const { container } = render(ui);
+  return createWrapper(container).findAutosuggest()!;
+}
+
+describe('Virtual scroll support', () => {
+  test('should render plain virtual list', () => {
+    const wrapper = renderWithWrapper(<Autosuggest {...defaultProps} />);
+    wrapper.findNativeInput().focus();
+    expect(wrapper.findDropdown().findOptions()).toHaveLength(3);
+  });
+
+  test('should render a subset of items in virtual list', () => {
+    const wrapper = renderWithWrapper(
+      <Autosuggest
+        {...defaultProps}
+        options={Array.from({ length: 100 }, (_, index) => ({ value: `Option ${index}` }))}
+      />
+    );
+    wrapper.findNativeInput().focus();
+    expect(wrapper.findDropdown().findOptions()).toHaveLength(6);
```

Comment:
how certain are we that this will remain `6`? isn't it somewhat dependent on a lot of things (3rd party implementation, item spacing, ...) that might change?

Reply:
This is defined by `overscan` configuration: https://github.com/cloudscape-design/components/blob/c95120a2acca52461e18575eda3aee4b34e2b937/src/autosuggest/virtual-list.tsx#L37

One current item + 5 overscan.

Reply:
Item size does not matter here, because it is a JSDOM test. It may only change if we upgrade `react-virtual` dependency, where changing this test would be totally fine.